src/3d/framegraph/qgs3daxisrenderview.cpp

```diff
@@ -0,0 +1,248 @@
+  mObjectCamera->setObjectName( objectName() + "::ObjectCamera" );
+  mObjectCamera->setProjectionType( cameraCtrl->camera()->projectionType() );
+  mObjectCamera->lens()->setFieldOfView( cameraCtrl->camera()->lens()->fieldOfView() * 0.5f );
+
+  Qt3DRender::QCameraSelector *cameraSelector = new Qt3DRender::QCameraSelector( objectFilter );
+  cameraSelector->setCamera( mObjectCamera );
+
+  // This ensures to have the label (Text2DEntity) rendered after the other objects and therefore
+  // avoid any transparency issue on the label.
+  Qt3DRender::QSortPolicy *objectSortPolicy = new Qt3DRender::QSortPolicy( cameraSelector );
+  QVector<Qt3DRender::QSortPolicy::SortType> objectSortTypes = QVector<Qt3DRender::QSortPolicy::SortType>();
+  objectSortTypes << Qt3DRender::QSortPolicy::BackToFront;
+  objectSortPolicy->setSortTypes( objectSortTypes );
+
+  Qt3DRender::QClearBuffers *objectClearBuffers = new Qt3DRender::QClearBuffers( objectSortPolicy );
+  objectClearBuffers->setBuffers( Qt3DRender::QClearBuffers::DepthBuffer );
+
+  // render pass for the axis label
+  // mTwoDLabelceneEntity->addComponent( twoDLayer );
+
+  Qt3DRender::QLayerFilter *labelFilter = new Qt3DRender::QLayerFilter( mViewport );
+  labelFilter->addLayer( mLabelLayer );
+
+  mLabelCamera = new Qt3DRender::QCamera;
+  mLabelCamera->setObjectName( objectName() + "::LabelCamera" );
+  mLabelCamera->setProjectionType( Qt3DRender::QCameraLens::ProjectionType::OrthographicProjection );
+
+  Qt3DRender::QCameraSelector *labelCameraSelector = new Qt3DRender::QCameraSelector( labelFilter );
+  labelCameraSelector->setCamera( mLabelCamera );
+
+  // this ensures to have the label (Text2DEntity) rendered after the other objects and therefore
+  // avoid any transparency issue on the label.
+  Qt3DRender::QSortPolicy *labelSortPolicy = new Qt3DRender::QSortPolicy( labelCameraSelector );
+  QVector<Qt3DRender::QSortPolicy::SortType> labelSortTypes = QVector<Qt3DRender::QSortPolicy::SortType>();
+  labelSortTypes << Qt3DRender::QSortPolicy::BackToFront;
+  labelSortPolicy->setSortTypes( labelSortTypes );
+
+  Qt3DRender::QClearBuffers *clearBuffers = new Qt3DRender::QClearBuffers( labelSortPolicy );
+  clearBuffers->setBuffers( Qt3DRender::QClearBuffers::DepthBuffer );
+
+  // update viewport size
+  onViewportSizeUpdate();
+}
+
+Qt3DRender::QViewport *Qgs3DAxisRenderView::viewport() const
+{
+  return mViewport;
+}
+
+Qt3DRender::QLayer *Qgs3DAxisRenderView::objectLayer() const
+{
+  return mObjectLayer;
+}
+
+Qt3DRender::QLayer *Qgs3DAxisRenderView::labelLayer() const
+{
+  return mLabelLayer;
+}
+
+Qt3DRender::QCamera *Qgs3DAxisRenderView::objectCamera() const
+{
+  return mObjectCamera;
+}
+
+Qt3DRender::QCamera *Qgs3DAxisRenderView::labelCamera() const
+{
+  return mLabelCamera;
+}
+
+void Qgs3DAxisRenderView::updateWindowResize( int width, int height )
+{
+  onViewportSizeUpdate( width, height );
+}
+
+
+void Qgs3DAxisRenderView::onViewportSizeUpdate( int width, int height )
```

Comment:
onViewportSizeUpdate() has default arguments `int width = -1, int height = -1` and onHorizontalPositionChanged()+onVerticalPositionChanged() call it with the default arguments - but the code inside that function does not seem to consider that - or does it?

Reply:
arrrrg... I started the work but did not end it :-1: